code/modules/organs/internal/carrion.dm

```diff
@@ -286,18 +287,48 @@
 	if(istype(food, /obj/item/weapon/grab))
 		var/obj/item/weapon/grab/grab = food
 		var/mob/living/carbon/human/H = grab.affecting
+		if (grab.state < GRAB_AGGRESSIVE)
+			to_chat(owner, SPAN_WARNING("Your grip upon [H] is too weak."))
+			return
 		if(istype(H))
 			var/obj/item/organ/external/E = H.get_organ(owner.targeted_organ)
+			if (tearing) // one at a time, thank you.
+				to_chat(owner, SPAN_WARNING("Your maw is already focused on something."))
+				return
+
 			if(E.is_stump())
 				to_chat(owner, SPAN_WARNING("You can't tear off a limb stump"))
 				return
+			tearing = TRUE
 
-			visible_message(SPAN_DANGER("\The [owner] bites into \the [H]'s \the [E] and starts tearing it off!"))
+			visible_message(SPAN_DANGER("[owner] bites into [H]'s [E.name] and starts tearing it [E.functions ? "off": "apart"]!"))
 			if(do_after(owner, 5 SECONDS, H))
+				tearing = FALSE
+				if (E.organ_tag == BP_HEAD)
+					if(!(H.incapacitated(INCAPACITATION_UNCONSCIOUS)))
+						var/datum/gender/G = gender_datums[H.gender]
+						for(var/obj/item/organ/external/smacker in H.organs)
+							if(smacker.functions == BODYPART_GRASP || smacker.functions == BODYPART_STAND)
+								if(!(smacker.is_broken() || smacker.is_dislocated() || smacker.limb_efficiency < 50))
+									owner.Stun(5) // I don't know how stun works, so magic number.
+									owner.drop_item()
+									visible_message(SPAN_DANGER("[H] smacks away [owner]'s maw with [G.his] [smacker.name]."))
+									return
+			
+				if (E.organ_tag == BP_GROIN || E.organ_tag == BP_CHEST) // torso bits should not be torn off.
+					E.take_damage(30)
+					var/obj/item/organ/internal/organ_to_remove = pick(E.internal_organs)
+					organ_to_remove?.loc = get_turf(src)
+					E.internal_organs -= organ_to_remove
+					visible_message(SPAN_DANGER("[owner] tears \a [organ_to_remove] out of [H]'s [E.name]!"))
+					playsound(loc, 'sound/voice/shriek1.ogg', 50)
```

Comment:
This already exists 3 lines below unless it doesn't proc

Reply:
if you had included one more line, you would have seen the return.

Reply:
me dum